tests/webservice_test.py

```diff
@@ -279,17 +279,27 @@ def test_unknown_error(self):
             self.run_client(self.client.country(ip))
 
     def test_request(self):
-        matcher = HeaderValueMatcher(
-            {
-                "Accept": "application/json",
-                "Authorization": "Basic NDI6YWJjZGVmMTIzNDU2",
-                "User-Agent": lambda x: x.startswith("GeoIP2-Python-Client/"),
-            }
-        )
+        def user_agent_compare(actual: str, expected: str) -> bool:
+            if actual is None:
+                return False
+            return actual.startswith("GeoIP2-Python-Client/")
+
+        def accept_compare(actual: str, expected: str) -> bool:
+            return actual == "application/json"
+
+        def authorization_compare(actual: str, expected: str) -> bool:
+            return actual == "Basic NDI6YWJjZGVmMTIzNDU2"
+
         self.httpserver.expect_request(
             "/geoip/v2.1/country/1.2.3.4",
             method="GET",
-            header_value_matcher=matcher,
+            header_value_matcher=HeaderValueMatcher(
+                {
+                    "User-Agent": user_agent_compare,
+                    "Accept": accept_compare,
+                    "Authorization": authorization_compare,
+                }
+            ),
```

Comment:
I don't think this works either. If I change the values, the tests don't fail. I was expecting something like this:

```
        def user_agent_compare(actual: str, expected: str) -> bool:
            if actual is None:
                return False
            return actual.startswith(expected)

        self.httpserver.expect_request(
            "/geoip/v2.1/country/1.2.3.4",
            method="GET",
            headers={
                    "Accept": "application/json",
                    "Authorization": "Basic NDI6YWJjZGVmMTIzNDU2",
                    "User-Agent": "GeoIP2-Python-Client/"
            },
            header_value_matcher=HeaderValueMatcher(defaultdict(
                lambda: HeaderValueMatcher.default_header_value_matcher,
                {"User-Agent": user_agent_compare}
            )),
        ).respond_with_json(
            self.country,
            status=200,
            content_type=self._content_type("country"),
        )
```

Alternatively, you could modify the default matcher as directed in the docs.